pkg/kubectl/cmd/proxy.go

```diff
@@ -70,13 +70,19 @@ The above lets you 'curl localhost:8001/custom/api/v1/pods'
 	cmd.Flags().String("accept-hosts", kubectl.DefaultHostAcceptRE, "Regular expression for hosts that the proxy should accept.")
 	cmd.Flags().String("reject-methods", kubectl.DefaultMethodRejectRE, "Regular expression for HTTP methods that the proxy should reject.")
 	cmd.Flags().IntP("port", "p", 8001, "The port on which to run the proxy.")
-	cmd.Flags().Bool("disable-filter", false, "If true, disable request filtering in the proxy. This is dangerous, and can leave you vulnerable to XSRF attacks.  Use with caution.")
+	cmd.Flags().StringP("unix", "u", "", "Unix socket on which to run the proxy.")
+	cmd.Flags().Bool("disable-filter", false, "If true, disable request filtering in the proxy. This is dangerous, and can leave you vulnerable to XSRF attacks, when used with an accessible port.")
 	return cmd
 }
 
 func RunProxy(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command) error {
+	path := cmdutil.GetFlagString(cmd, "unix")
 	port := cmdutil.GetFlagInt(cmd, "port")
-	fmt.Fprintf(out, "Starting to serve on localhost:%d", port)
+	if path == "" {
+		fmt.Fprintf(out, "Starting to serve on localhost:%d\n", port)
+	} else {
+		fmt.Fprintf(out, "Starting to serve on %s\n", path)
```

Comment:
Please generate an error if port is set to a non-default value and --unix-socket is also specified.